src/django_pg_migration_tools/operations.py

```diff
@@ -32,7 +32,7 @@ def safer_create_index(
         to_state: migrations.state.ProjectState,
         index: models.Index,
         model: type[models.Model],
-        operation: SaferAddIndexConcurrently,
+        operation: SaferAddIndexConcurrently | SaferRemoveIndexConcurrently,
```

Comment:
can this instead be a Protocol class as the type you expect?

Reply:
There should be the only two classes that create/drop indexes, so I'm not sure there're gains in using structural typing here if we're not aiming for flexibility. On the side, nominal typing provides stronger contracts too.